logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/PersistMessageUseCase.kt

```diff
@@ -0,0 +1,49 @@
+package com.wire.kalium.logic.data.message
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.onSuccess
+
+/**
+ * Internal UseCase that should be used instead of MessageRepository.persistMessage(Message)
+ * It automatically updates ConversationModifiedDate and ConversationNotificationDate if needed
+ */
+interface PersistMessageUseCase {
+    suspend operator fun invoke(message: Message): Either<CoreFailure, Unit>
+}
+
+class PersistMessageUseCaseImpl(
+    private val messageRepository: MessageRepository,
+    private val conversationRepository: ConversationRepository,
+    private val userId: QualifiedID,
+) : PersistMessageUseCase {
+
+    override suspend operator fun invoke(message: Message): Either<CoreFailure, Unit> {
+        return messageRepository.persistMessage(message)
+            .onSuccess {
+                if (message.content.shouldUpdateConversationOrder())
+                    conversationRepository.updateConversationModifiedDate(message.conversationId, message.date)
+
+                if (userId == message.senderUserId)
+                    conversationRepository.updateConversationNotificationDate(message.conversationId, message.date)
+            }
+    }
+
+    private fun MessageContent.shouldUpdateConversationOrder(): Boolean =
+        when (this) {
+            is MessageContent.MemberChange -> false
```

Comment:
```suggestion
            is MessageContent.MemberChange -> true
```

As I remember from wolfgang the `MemberChange` messages should update the conversation list order